### ISSUE TITLE
Compilation errors due to fixed typs in deal.II

### ISSUE THREAD

**Issue by jh66637** [exadg/exadg#252]
Explicitly using ```double``` for interface coupling results in compilation errors after the fixture of types in deal.II https://github.com/dealii/dealii/pull/14142.

The decision to use ```double``` instead of ```Number``` is related due to the design choice to keep ```Number``` out of ```BoundaryDescriptor```s. 

I might find some time to work on this towards the end of the week.

**Comment by peterrum** [comment]
A minimal test:
```cpp
#include <deal.II/matrix_free/fe_point_evaluation.h>

using namespace dealii;

template <int n_components, int dim, int spacedim, typename Number>
void
test()
{
  std::unique_ptr<Mapping<dim, spacedim>>       mapping;
  std::unique_ptr<FiniteElement<dim, spacedim>> fe;

  FEPointEvaluation<n_components, dim, spacedim, Number> fpe(
    *mapping, *fe, UpdateFlags::update_default);

  Triangulation<dim, spacedim> tria;

  fpe.reinit(tria.begin(), ArrayView<const Point<dim>>());

  fpe.evaluate(ArrayView<const Number>(), EvaluationFlags::values);

  fpe.integrate(ArrayView<Number>(), EvaluationFlags::values);
}

int
main()
{
  test<1, 1, 1, double>();
  test<2, 1, 1, double>();

  test<1, 2, 2, double>();
  test<2, 2, 2, double>();
  test<3, 2, 2, double>();

  test<1, 3, 3, double>();
  test<2, 3, 3, double>();
  test<3, 3, 3, double>();
  test<4, 3, 3, double>();

  test<1, 1, 1, float>();
  test<2, 1, 1, float>();

  test<1, 2, 2, float>();
  //test<2, 2, 2, float>(); // not working 
  test<3, 2, 2, float>();

  test<1, 3, 3, float>();
  test<2, 3, 3, float>();
  //test<3, 3, 3, float>(); // not working
  test<4, 3, 3, float>();
}
```

**Comment by jh66637** [comment]
Thank you :)

**Comment by nfehn** [comment]
should be resolved by  dealii/dealii#14159
